synapse/crypto/keyring.py

```diff
@@ -511,7 +511,7 @@ async def _fetch_keys(
             for key_id in queue_value.key_ids
         )
 
-        res = await self.store.get_server_keys_json(key_ids_to_fetch)
+        res = await self.store.get_server_signature_keys(key_ids_to_fetch)
```

Comment:
This essentially reverts #15417 and will cause large amounts of federation traffic. See #15171.

Reply:
Oh boo, for some reason I thought that the both perspective and direct key fetcher persisted in there :facepalm:

Reply:
Opposite, see https://github.com/matrix-org/synapse/issues/15171#issuecomment-1464224490 they both store into `server_keys_json`.